app/services/exports/export_report_csv_service.rb

```diff
@@ -31,5 +31,22 @@ def generate_csv_data
 
       csv_data
     end
+
+    def generate_range_csv_data
+      csv_data = []
+      return csv_data if @reports.empty?
+
+      headers = @reports.first.all_reports.flat_map { |r| r['entries'].keys }
+      csv_data << ['Year'] + headers
+
+      @reports.each do |report|
+        report_data = report.all_reports
+        year = report['year']
+        values = report_data.flat_map { |r| r['entries'].values }
+        csv_data << [year] + values
+      end
```

Comment:
Hi! Different reports end up having different columns. Here, instead of plucking out the values in the specific order, can you instead do a pass through all the reports and get all the headers and then build out a hash or something so that we can handle blank values and then go through and output the CSV based on that?

Something like this (note this is COMPLETELY UNTESTED SUGGESTION) :)

```suggestion
      @reports.each do |report|
        report_data = report.all_reports
        year = report['year']
        values = report_data.flat_map { |r|
          headers.map { |field| r['entries'][field] }
        }
        csv_data << [year] + values
      end
```

That way if year 1 has some headers that year 2 doesn't have (or vice-versa) then we will get all of the entries.